File: JavaScript/Clock.js
function Clock(indicatedHour, indicatedMin, extraTime) {
    let actualHour = 0;
    let actualMin = 0;
    let hour = 0;

    if(indicatedHour<0 || indicatedMin<0)
        return false;

    while(actualHour<indicatedHour) {
        if((actualMin+extraTime)<60) {
            actualHour++;
            actualMin += extraTime;
            hour++;
        } else if((actualMin+extraTime)>=60) {
            actualHour += 2;
            actualMin = (actualMin+extraTime)-60;
            hour++;
        }
    }
    return hour;
}

function testFunction(indicatedHour, indicatedMin, extraTime, expectedResult)
{
	if(Clock(indicatedHour, indicatedMin, extraTime)===expectedResult)
		return 'Pass';
	else	
		return 'Not pass';
}

console.log('Test Cases:');
console.log('\nIndicated Time\tActual Result\tExpected Result\tTest Case');
console.log('9:20\t\t'+Clock(9, 20, 10)+'\t\t'+8+'\t\t'+testFunction(9, 20, 10, 8));
console.log('12:30\t\t'+Clock(12, 30, 15)+'\t\t'+10+'\t\t'+testFunction(12, 30, 15, 10));
console.log('-12:30\t\t'+Clock(-12, 30, 15)+'\t\t'+false+'\t\t'+testFunction(-12, 30, 15, false));

File: ProgrammingBasics/Clock.c
#include<stdio.h>

int clock(int hour, int minutes, int x) 
{
	int actualHour = 0;
	
	while(hour>0)
	{
		if(minutes-x<0)
		{
			minutes = 60-x;
			hour = hour-2;
			actualHour++;
		}
		else
		{
			minutes = minutes-x;
			hour = hour-1;
			actualHour++; 
		}
	}
	return actualHour;
}

const char* testFunction(int hour, int minutes, int x, int expectedResult) 
{
    if(clock(hour,minutes,x)==expectedResult)
        return "Pass";
    else   
        return "Not pass";
}

int main()
{
	printf("Test Cases:\n");
    printf("\nIndicated Time\tActual Result\tExpected Result\tTest Case");
	printf("\n9:20\t\t%d\t\t8\t\t%s",clock(9,20,10),testFunction(9,20,10,8));
	printf("\n12:30\t\t%d\t\t10\t\t%s",clock(12,30,15),testFunction(12,30,15,10));
	printf("\n8:45\t\t%d\t\t7\t\t%s",clock(8,45,15),testFunction(8,45,15,7));
	printf("\n10:30\t\t%d\t\t9\t\t%s",clock(10,30,10),testFunction(10,30,10,9));
}

